component/client/chat/ClientIndependentChatRoom.tsx

```diff
@@ -22,27 +23,24 @@ class ClientIndependentChatRoom extends React.Component<any, IClientIndependentC
     super();
     this.state = {
       currentUrl: Constants.NOOP_URL,
+      userID: Constants.NOOP_ID,
       username: Constants.NOOP_USERNAME
     };
-    storage.get("username")
-      .then((username) => {
-        if (username) {
+    storage.get(Constants.STORAGE_KEY_USERNAME).then((username) => {
+      storage.get(Constants.STORAGE_KEY_ID).then((userID) => {
```

Comment:
If React does optimize, 2 renders (initial and data loaded are coupled together). In the worst case, we have 3 renders (inital, username, and id). The errors that are thrown occasionally is when username is provided; however, id is not because React did not optimize for this or vice versa. The initialization already puts us into a NOOP state. We should only re-render again if it can fully gets out rather than partially get us out.